src/sidebar/util/groups.js

```diff
@@ -19,9 +41,22 @@ export function combineGroups(userGroups, featuredGroups, uri) {
   const myGroupIds = userGroups.map(g => g.id);
   featuredGroups = featuredGroups.filter(g => !myGroupIds.includes(g.id));
 
+  // Set the `canLeave` property. Groups can be left if they are private unless
+  // the global `allowLeavingGroups` value is false in the config settings object.
+  const setCanLeave = group => {
+    group.canLeave = !allowLeavingGroups(settings)
+      ? false
+      : group.type === 'private';
+  };
   // Set the isMember property indicating user membership.
-  featuredGroups.forEach(group => (group.isMember = false));
-  userGroups.forEach(group => (group.isMember = true));
+  featuredGroups.forEach(group => {
```

Comment:
I had to leave this separate because it looks like `isMember` gets set to false for feature and true otherwise. If i merged them first, I don't think it would be possible to know the difference w/o baking it meta data or doing a lookup each iteration of the loop. Thoughts?

Reply:
Ah, gotcha. But perhaps something like this?

```
featuredGroups.forEach(group => group.isMember = false);
userGroups.forEach(group => group.isMember = true);

const groups = userGroups.concat(featuredGroups);

groups.forEach(group => setCanLeave(group));
```

Reply:
No prob